src/services/neo4jService.js

```diff
@@ -1,185 +1,172 @@
+import { driver as createDriver } from "neo4j-driver";
 
-/**
- * Hooks into the neo4jDesktopApi
- *
- * Note: this integration is going to be deprecated in desktop api 2.0
- */
-const neo4jDesktopApi = window.neo4jDesktopApi;
+class Neo4JService {
+  _getNeo4jDriver = async () => {
+    if (!this.driver) {
+      try {
+        /**
+         * Hooks into the neo4jDesktopApi.
+         * Note: this integration is going to be deprecated in desktop api 2.0
+         */
+        const context = await window.neo4jDesktopApi.getContext();
 
-/**
- * Returns a neo4j driver instance from the boltUrl, username, and passwords.
- *
- * Currently accepts bolt and bolt+routing
- * @param {String} boltUrl
- * @param {String} username
- * @param {String} password
- */
-const createDriver = (boltUrl, username, password) =>
-  neo4j.driver(boltUrl, neo4j.auth.basic(username, password));
-
-export default {
-  getNeo4jDriver: async function () {
-    let driver = undefined;
-    if (neo4jDesktopApi) {
-      await neo4jDesktopApi.getContext().then((context) => {
         for (let project of context.projects) {
           for (let graph of project.graphs) {
             if (graph.status === "ACTIVE") {
-              console.log(
-                "Active graph is; " +
-                  graph.name +
-                  " (" +
-                  graph.description +
-                  ")"
-              );
+              console.log(`Active graph is: ${graph.name} - (${graph.description})`);
+
               let boltProtocol = graph.connection.configuration.protocols.bolt;
-              driver = createDriver(
+
+              // Already found and athenticated an active graph.
+              // No need to try and authenticate other graphs.
+              this.driver = createDriver(
                 boltProtocol.url,
                 boltProtocol.username,
                 boltProtocol.password
               );
             }
           }
         }
-      });
+      } catch (error) {
+        console.log(error);
+      }
     }
-    return driver;
-  },
 
-  getNodeLabels: async function (driver) {
-    if (driver === undefined) return [];
+    return this.driver;
+  };
+
+  _runQuery = async (query, params = undefined) => {
+    const driver = this.driver || await this._getNeo4jDriver();
+
+    if (!driver) {
+      throw new Error('Failed to get driver');
+    }
 
-    let res = [];
     const session = driver.session();
-    await session
-      .run(`CALL db.labels() YIELD label RETURN label ORDER BY label`)
-      .then(function (result) {
-        result.records.forEach(function (record) {
-          let el = {
-            value: record.get("label"),
-            label: record.get("label"),
-          };
-          res.push(el);
-        });
-        session.close();
-      })
-      .catch(function (error) {
-        console.log(error);
-      });
 
-    return res;
-  },
+    const records = await session.run(query, params).then(
+      records => records,
```

Comment:
We should return `records.records` here, to be consistent with the previous behavior.

Reply:
This returns records. The callback, `records => records` is shorthand for `records => { return records }`